### DISCUSSION TITLE
API: syntax for `listOptions.labelSelector` and `fieldSelector`

### DISCUSSION THREAD

**Discussion by Lipphi** [argoproj/argo-workflows#8500] [Q&A] [answered]
Hi,
Most methods of the API provide optional parameters  `listOptions.labelSelector` and `listOptions.fieldSelector`.
I wondered :
* what is the distinction between fields and labels? 
* which fields are accessible ?
* are comparison operators (`<`,`>`,`!=`,...) and regex (or some contains method implicitly applied) accepted ?
* finally, if there's already a documentation summing it all up , with the syntax to use , that would be great !

My concern is:
I manage to get some simple filters with a workflow name, for instance, this requests works:
`https://argoworkflow.mydomain.com/api/v1/workflows/my-namespace?listOptions.fieldSelector=metadata.name=hello-world-template-cl9mn`
Then I'd like to filter by status phase , or by start date, but these don't work:
1. Status:
    *  Request: `https://argoworkflow.mydomain.com/api/v1/workflows/my-namespace?listOptions.fieldSelector=status.phase=Succeeded`

    * Response: `{
    "code": 9,
    "message": "field label not supported: status.phase"
}`

2. startedAt: (from my naive guessing...)
    *  Request: `https://argoworkflow.mydomain.com/api/v1/workflows/my-namespace?istOptions.fieldSelector=status.startedAt<2022-04-05T00:00:0Z`

    * Response: `{
    "code": 9,
    "message": "invalid selector: 'status.startedAt<2022-04-05T00:00:0Z'; can't understand 'status.startedAt<2022-04-05T00:00:0Z'"
}`

Thanks in advance,
Philippe

**Comment by terrytangyuan** [comment, marked as answer]
Not all fields are supported. See https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/#supported-fields

Also unlike label selector, only limited operators are supported for field selector: https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/#supported-operators

**Reply by Lipphi** [reply]
Thanks for the reply.

**Reply by azazel75** [reply]
But, how can I see which are the supported fields for the Workflow resource? Trying with a non supported field doesn't give back the list ofd those supported ad in the examples on kubernetes.io

**Reply by agilgur5** [reply]
@azazel75 Unfortunately, only `metadata.name` and `metadata.namespace` are currently supported for field selectors for Custom Resources. This is a current Kubernetes limitation, not specific to Argo: https://github.com/kubernetes/kubernetes/issues/53459

As a result, some projects use labels and label selectors as a workaround. Argo manages some convenience labels on Workflows as well.

**Reply by azazel75** [reply]
Thanks for the reference!
